js/adapt-contrib-branching.js

```diff
@@ -1,287 +1,144 @@
-      branching[id] = Adapt.offlineStorage.serialize(trackingIds);
-      Adapt.offlineStorage.set('b', branching);
-    }
+  getSubsetByModelId(modelId) {
+    return this._rawSets.find(set => set.model.get('_id') === modelId);
+  }
 
-    /**
-     * Fetch an array representing the relative location of nextModel to the nearest tracking id.
-     * @param {Backbone.Model} nextModel
-     * @returns {Array<Number, Number>}
-     */
-    getTrackingPositionFromModel(nextModel) {
-      const firstDescendant = nextModel.getAllDescendantModels(false).concat([nextModel])[0];
-      const nearestTrackingIdModel = [firstDescendant].concat(firstDescendant.getAncestorModels()).find(model => model.has('_trackingId'));
-      const trackingId = nearestTrackingIdModel.get('_trackingId');
-      const trackingIdGroup = [nearestTrackingIdModel].concat(nearestTrackingIdModel.getAllDescendantModels(true));
-      const indexInTrackingIdGroup = trackingIdGroup.findIndex(descendant => descendant === nextModel);
-      if (indexInTrackingIdGroup >= 0) {
-        // Is either the nearestTrackingIdModel (0) or one of its flattened descendants (>0)
-        return [ trackingId, indexInTrackingIdGroup ];
-      }
-      // Is an ancestor of the nearestTrackingIdModel
-      const trackingIdAncestors = nearestTrackingIdModel.getAncestorModels();
-      const ancestorDistance = trackingIdAncestors.findIndex(ancestor => ancestor === nextModel);
-      return [ trackingId, -ancestorDistance ];
-    }
+  setupEventListeners() {
+    this.listenTo(Adapt, 'view:requestChild', this.onRequestChild);
+    this.listenTo(data, 'change:_isComplete', this.onComplete);
+  }
 
-    onPopupOpened() {
-      this.openPopupCount++;
-    }
+  onRequestChild(event) {
+    const set = this.getSubsetByModelId(event.target.model.get('_id'));
+    if (!set) return;
+    const nextModel = set.getNextModel();
+    if (nextModel === false) {
+      // Previous model isn't complete
+      return;
+    }
+    if (nextModel === true) {
+      // No further models, manually complete branching container
+      event.target.model.setCompletionStatus();
+      return;
+    }
+    const clonedModel = set.addNextModel(nextModel);
+    event.model = clonedModel;
+    // Stop rendering children in this parent to allow for user responses
+    event.stopNext();
+  }
 
-    onPopupClosed() {
-      this.openPopupCount--;
-      if (!this.shouldContinueOnPopupClose) return;
-      if (this.openPopupCount > 0) return;
-      this.shouldContinueOnPopupClose = false;
-      this.continue();
-    }
+  onPopupOpened() {
+    this.openPopupCount++;
+  }
 
-    async continue() {
-      if (this.openPopupCount) {
-        this.shouldContinueOnPopupClose = true;
-        return;
-      }
-      await Adapt.parentView.addChildren();
-    }
+  onPopupClosed() {
+    this.openPopupCount--;
+    if (!this.shouldContinueOnPopupClose) return;
+    if (this.openPopupCount > 0) return;
```

Comment:
could do this instead of two if statements
`if (!this.shouldContinueOnPopupClose || this.openPopupCount > 0) return;`